rundeckapp/grails-app/controllers/rundeck/controllers/ApiController.groovy

```diff
@@ -206,62 +211,115 @@ class ApiController extends ControllerBase{
         forward(uri: servletPath.replace('/*', "/$name"))
     }
 
-    /**
-     * API endpoints
-     */
-    /**
-     * Return true if grails configuration allows given feature, or '*' features
-     * @param name
-     * @return
-     */
-    private boolean featurePresent(def name){
-        boolean featureStatus = configurationService.getBoolean("feature.incubator.${name}", false)
 
-        def splat=configurationService.getBoolean("feature.incubator.*", false)
-        return splat || featureStatus
-    }
+
+
+
+    @Tag(name = "system")
     /**
-     * Set an incubator feature toggle on or off
-     * @param name
-     * @param enable
+     * API endpoint to query system features' toggle status: True/False for On/Off
+     *
+     * The URL is `/feature/{featureName}` where featureName is the specific name of the feature without `rundeck.feature.` prefix and `enabled` surfix.
+     * E.g. The configuration item name of the feature to enable runner is `rundeck.feature.runner.enabled`, to query the status of this feature
+     *      the request will be `/feature/runner`. The result of this query is a JSON object { name: "$featureName", "enabled": true/false }
+     *
      */
-    private void toggleFeature(def name, boolean enable){
-        grailsApplication.config.feature?.incubator?.putAt(name, enable)
+    @Get(
+        uri= "/feature/{featureName}",
+        produces = MediaType.APPLICATION_JSON
+    )
+    @Operation(
+        method = "GET",
+        summary = "Get Rundeck System Feature Status",
+        description = "The result of this query is a JSON object { name: \"featureName\", \"enabled\": true/false }",
+        parameters = [
+            @Parameter(
+                    name='featureName',
+                    in = ParameterIn.PATH,
+                    description = 'Feature name without the `feature.` prefix, or blank to receive list of all system features',
+                    allowEmptyValue = true,
+                    required = false,
+                    schema = @Schema(
+                            type='string'
+                    )
+            )
+        ],
+        responses = [
+            @ApiResponse(
+                responseCode = "200",
```

Comment:
OpenAPI annotation cannot have two ApiResponse with the same `responseCode`, the last one will be used to generate the document. So I split the API into two: single feature query and list all features.

Reply:
that's fine to do it that way.

btw, I think you can do it like this within one response code `@Schema(oneOf=[Type1,Type2])`